src/main/kotlin/com/cognifide/gradle/aem/instance/tasks/Tail.kt

```diff
@@ -0,0 +1,102 @@
+package com.cognifide.gradle.aem.instance.tasks
+
+import com.cognifide.gradle.aem.common.AemDefaultTask
+import com.cognifide.gradle.aem.common.Formats
+import com.cognifide.gradle.aem.instance.Instance
+import com.cognifide.gradle.aem.instance.tail.*
+import com.cognifide.gradle.aem.instance.tail.io.FileDestination
+import com.cognifide.gradle.aem.instance.tail.io.LogFiles
+import com.cognifide.gradle.aem.instance.tail.io.UrlSource
+import kotlin.math.max
+import kotlinx.coroutines.ObsoleteCoroutinesApi
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
+import org.gradle.api.tasks.Nested
+import org.gradle.api.tasks.TaskAction
+import org.gradle.process.internal.shutdown.ShutdownHooks
+
+/**
+ * TODO: Since coroutines API is still in experimental mode we would need to adapt to it's final API when released.
+ * Please see https://github.com/Kotlin/kotlinx.coroutines/issues/632#issuecomment-425408865
+ */
+@UseExperimental(ObsoleteCoroutinesApi::class)
+open class Tail : AemDefaultTask() {
+
+    init {
+        description = "Tails logs from all configured instances (local & remote) and notifies developer about unknown errors."
+    }
+
+    private val logFileCreator = LogFiles(aem, name)
+
+    @Nested
+    private val config = TailConfig()
+
+    fun isRunning() = logFileCreator.isLocked()
+
+    fun config(configurer: TailConfig.() -> Unit) {
+        config.apply(configurer)
+    }
+
+    @TaskAction
+    fun tail() {
+        var shouldRunTailing = true
+        ShutdownHooks.addShutdownHook {
+            shouldRunTailing = false
+        }
+        if (logFileCreator.isLocked()) {
+            logger.warn(
+                "Another instance of aemTail ($name) is running for this project. " +
```

Comment:
avoid using task names in messages, because this will be not true if somebody decide to extend this task

Reply:
OK